packages/scene-composer/src/components/panels/scene-settings/SceneTagSettingsEditor.tsx

```diff
@@ -37,62 +36,38 @@ export const SceneTagSettingsEditor: React.FC = () => {
           ...getSceneProperty(KnownSceneProperty.ComponentSettings),
           [KnownComponentType.Tag]: newTagSettings,
         };
-
         setSceneProperty(KnownSceneProperty.ComponentSettings, newComponentSettings);
       }
       setTagSettings(newTagSettings);
     },
-    [tagSettings, getSceneProperty, setSceneProperty],
+    [internalScale, tagSettings, getSceneProperty, setSceneProperty],
   );
 
   const onSliderChange = useCallback(
     (event) => {
       setInternalScale(Number(event.target.value));
-      setDirty(true);
     },
-    [setInternalScale, setDirty],
+    [internalScale, tagSettings],
   );
 
-  const onInputBlur = useCallback(() => {
-    // If slider is getting focus, this makes sure to execute setFocusSlider(true) first to keep showing it.
-    const id = setTimeout(() => {
-      setFocusInput(false);
-    }, 1);
-    return () => {
-      clearTimeout(id);
-    };
-  }, [setFocusInput]);
-
   const onInputChange = useCallback(
     (event) => {
       let value = event.detail.value;
       if (value < 0) {
         value = 0;
       }
-
       if (value !== internalScale) {
         setInternalScale(Number(value));
-        setDirty(true);
       }
     },
-    [setInternalScale, setDirty],
+    [internalScale, tagSettings],
   );
 
-  // Save scale changes to settings
   useEffect(() => {
-    // In viewing mode, dragging slider will update component immediately.
-    if (dirty && !focusInput && (!draggingSlider || isViewing)) {
+    if (internalScale !== tagSettings.scale) {
       updateSettings({ scale: internalScale });
-      setDirty(false);
-    }
-  }, [updateSettings, setDirty, dirty, focusInput, internalScale, draggingSlider, isViewing]);
-
-  // Update internal when scale in store is changed
-  useEffect(() => {
-    if (!dirty && tagSettings.scale !== internalScale) {
-      setInternalScale(tagSettings.scale);
     }
-  }, [dirty, draggingSlider, tagSettings.scale, internalScale]);
+  }, [internalScale, tagSettings]);
```

Comment:
I don't think you need this.

"If internal scale doesn't equal scale, set it to scale" is the same as "set internal scale to scale". You don't need a condition, therefore you don't need a check, and you don't need the useEffect at all.

Reply:
This monitors `internalScale` for updates & only triggers the update to settings on scene properties if there is an appropriate change to be made. This covers updates via input, slider, & keystroke. It is necessary to keep them all in sync.